pkg/backends/influxdb/flux/parser.go

```diff
@@ -0,0 +1,123 @@
+package flux
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/trickstercache/trickster/v2/pkg/timeseries"
+	"github.com/trickstercache/trickster/v2/pkg/util/timeconv/duration"
+)
+
+type Parser struct {
+	reader io.Reader
+}
+
+func NewParser(reader io.Reader) *Parser {
+	return &Parser{
+		reader: reader,
+	}
+}
+
+func (p *Parser) ParseQuery() (*Query, error) {
+	valid := false
+	r := bufio.NewReader(p.reader)
+	ln := 0
+	q := &Query{}
+	for line, err := r.ReadString('\n'); ; line, err = r.ReadString('\n') {
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			} else {
+				return nil, err
+			}
+		}
+		// Check for line 0 "from"
+		if ln == 0 {
+			if !strings.Contains(line, "from") {
+				return nil, FluxSyntax(line, "flux scripts must begin with from(bucket: ...)")
+			}
+		}
+		// Check for range
+		if strings.Contains(line, "range") {
+			q.Extent, err = parseRangeFilter(line)
+			if err != nil {
+				return nil, err
+			}
+			valid = true
+		}
+		q.Statement += line + " "
+		ln++
+	}
+	if !valid {
+		return nil, FluxSemantics("script is not valid flux")
+	}
+	return q, nil
+}
+
+// Parse a line that is a range filter range(start: $[start], stop: $[stop])
+func parseRangeFilter(line string) (timeseries.Extent, error) {
+	tokens := strings.FieldsFunc(line, func(r rune) bool {
+		return r == ' ' || r == '(' || r == ')' || r == ','
+	})
+	var start, stop time.Time
+	var ok bool
+	for i, token := range tokens {
+		if token == "start:" {
+			start, ok = tryParseTimeField(tokens[i+1])
+			if !ok {
+				return timeseries.Extent{}, FluxSyntax(token, "must be a valid duration literal, RFC3339 time, or UNIX time")
+			}
+		}
+		if token == "stop:" {
+			stop, ok = tryParseTimeField(tokens[i+1])
+			if !ok {
+				return timeseries.Extent{}, FluxSyntax(token, "must be a valid duration literal, RFC3339 time, or UNIX time")
+			}
+		}
+	}
+	if start.IsZero() {
+		return timeseries.Extent{}, FluxSemantics("range() expressions require a valid start argument")
+	}
+	return timeseries.Extent{Start: start, End: stop}, nil
+}
+
+func tryParseTimeField(s string) (t time.Time, ok bool) {
```

Comment:
Let's replace the bool return vals in these functions with error. since the 3 `try` subfunctions actually check for errors and only return false in the event of an error, we should just pass the original error all the way back through to the calling function (parseRangeFilter), esp since parseRangeFilter _already_ passes an error back to its calling function (ParseQuery). No need to go from err > bool > custom-created error. Remember, in applications like trickster that can handle tens of thousands of requests per second, minimizing memory churn is important to performance.

Also, since the rest of the codebase does _not_ use named response variables in the prototype, we should maintain that stylistic choice across this new package.